src/main/resources/drivers/Arbor_ArbOS.js

```diff
@@ -0,0 +1,224 @@
+	},
+	"configurationCommitted": {
+		type: "Binary",
+		title: "Configuration committed",
+		searchable: true
+	}
+};
+
+var CLI = {
+	ssh: {
+		macros: {
+			exec: {
+				options: ["exec"],
+				target: "exec"
+			}
+		},
+		pager: {
+			match: /^\[more\]$/,
+			response: " "
+		},
+	},
+	exec: {
+		prompt: /^([A-Za-z0-9\.\-_]+@[A-Za-z\-_0-9\.]+):\/#/,
+	}
+};
+
+function snapshot(cli, device, config) {
+	let systemHardwareCleanup = function (config) {
+		config = config.replace(/^Boot time.*$\n/mg, "");
+		config = config.replace(/^Load averages.*$\n/mg, "");
+		return config;
+	};
+
+	cli.macro("exec");
+
+	// Device type
+	device.set("networkClass", "SERVER");
+	device.set("family", "Arbor ArbOS");
+
+	// System version
+	let systemVersionResult = cli.command("system version");
+	systemVersion = systemVersionResult.match(/Version: (.*)/m);
+	if (systemVersion != null) {
+		config.set("systemVersion", systemVersion[1]);
+		device.set("softwareVersion", systemVersion[1]);
+	}
+
+	// System packages
+	let systemPackagesResult = cli.command("system files show");
+	config.set("systemPackages", systemPackagesResult);
+
+	// System hardware
+	let systemHardwareResult = systemHardwareCleanup(cli.command("system hardware").replace(/\r\n/g, "\n"));
+	config.set("systemHardware", systemHardwareResult);
+
+	// Total memory size
+	let memoryPattern = /Memory Device: (.*) MB .*/mg;
+	let memory = 0;
+	let memoryMatch;
+	while (memoryMatch = memoryPattern.exec(systemHardwareResult)) {
+		memory += parseInt(memoryMatch[1]);
+	}
+	device.set("memorySize", memory);
+
+	// Serial number
+	let serialMatch = systemHardwareResult.match(/Serial Number: (.*)/m);
+	if (serialMatch != null) {
+		device.set("serialNumber", serialMatch[1]);
+	}
+
+	// System configuration
+	let systemConfiguration = cli.command("config show");
+	config.set("systemConfiguration", systemConfiguration);
+
+	// Hostname
+	let hostname = systemConfiguration.match(/^system name set (.+)$/m);
+	if (hostname != null) {
+		device.set("name", hostname[1]);
+	}
+
+	// System location information
+	let location = systemConfiguration.match(/^services sp device edit .* snmp location set (.+)$/m);
+	if (location != null) {
+		device.set("location", location[1]);
+	}
+
+	// System contact information
+	let contact = systemConfiguration.match(/^services sp preferences support_email set (.+)$/m);
+	if (contact != null) {
+		device.set("contact", contact[1]);
+	}
+
+	// Check if the configuration has been committed (check for existing diff)
+	let configDiffResult = cli.command("config diff").replace(/\r\n/g, "\n");
+	if (configDiffResult.split("\n").length === 1) {
+		device.set("configurationCommitted", true);
+	} else {
+		device.set("configurationCommitted", false)
+	}
+
+	// Gather network interfaces information
+	let interfaces = cli.command("ip interfaces show").replace(/\r\n/g, "\n").trim().split("\n\n");
+	for (let interface of interfaces) {
+		let ifName = interface.split(" ")[0];
```

Comment:
Should rather use `const` for JS variables which won't change

Reply:
const is not only for statically defined variables ?

Reply:
Fixed

Reply:
Variables like `ifName` are scoped to the `for` loop, and are not changed from their initial value, so should be defined with `const`.

Reply:
Done